src/fugw/solvers/dense.py

```diff
@@ -126,7 +129,7 @@ def fugw_loss(
         loss = 0
 
         if alpha != 1 and D is not None:
-            loss_wasserstein = (D * pi).sum() + (D * gamma).sum()
+            loss_wasserstein = ((D * pi).sum() + (D * gamma).sum()) / 2
```

Comment:
Sorry, I actually introduced this error in a very recent PR: https://github.com/alexisthual/fugw/pull/28

I initially thought this factor 2 was a mistake. Actually, it is not: it is present because our cost function is derived from a formulation of the FUGW loss in which we seek to find two plans P and Q. They both linearly contribute to the half of the wasserstein component (the very line you commented shows that the wasserstein component can be interpreted as the wasserstein loss wrt to (P+Q)/2), and interact in the gromov component. Hence the presence of a 1/2 coefficient when deriving the wasserstein component with regard to P with Q being fixed.